xbmc/utils/test/TestArtUtils.cpp

```diff
@@ -7,19 +7,75 @@
  */
 
 #include "FileItem.h"
+#include "URL.h"
+#include "filesystem/Directory.h"
 #include "platform/Filesystem.h"
 #include "utils/ArtUtils.h"
 #include "utils/FileUtils.h"
+#include "utils/StringUtils.h"
 #include "utils/URIUtils.h"
+#include "video/VideoInfoTag.h"
 
 #include <array>
 #include <fstream>
+#include <random>
 
 #include <fmt/format.h>
 #include <gtest/gtest.h>
 
 using namespace KODI;
 
+namespace
+{
+
+std::string unique_path(const std::string& input)
+{
+  std::random_device rd;
+  std::mt19937 gen(rd());
+  auto randchar = [&gen]()
+  {
+    const std::string set = "0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ";
+    std::uniform_int_distribution<> select(0, set.size() - 1);
+    return set[select(gen)];
+  };
+
+  std::string ret;
+  ret.reserve(input.size());
+  std::transform(input.begin(), input.end(), std::back_inserter(ret),
+                 [&randchar](const char c) { return (c == '%') ? randchar() : c; });
+
+  return ret;
+}
+
+struct FanartTest
+{
+  std::string path;
+  std::string result;
+};
+
+class GetLocalFanartTest : public testing::WithParamInterface<FanartTest>, public testing::Test
+{
+};
+
+const auto local_fanart_tests = std::array{
+    FanartTest{"stack://#DIRECTORY#/foo-cd1.avi , #DIRECTORY#/foo-cd2.avi", "foo-fanart.jpg"},
+    FanartTest{"stack://#DIRECTORY#/foo-cd1.avi , #DIRECTORY#/foo-cd2.avi", "foo-cd1-fanart.jpg"},
```

Comment:
Windows test failure is related to backslash but not sure why only fails the first two. Changing to this test pass (on Windows):

```
    FanartTest{"stack://#DIRECTORY#\\foo-cd1.avi , #DIRECTORY#\\foo-cd2.avi", "foo-fanart.jpg"},
    FanartTest{"stack://#DIRECTORY#\\foo-cd1.avi , #DIRECTORY#\\foo-cd2.avi", "foo-cd1-fanart.jpg"},
```

Temp dir on Windows is returned with backslash style then seems the fail is due mix slash and backslash in the path.